circup/__init__.py

```diff
@@ -409,6 +410,616 @@ def __repr__(self):
         )
 
 
+class Backend:
+    """
+    Backend parent class to be extended for workflow specific
+    implementations
+    """
+
+    def __init__(self):
+        self.LIB_DIR_PATH = None
+
+    def _get_circuitpython_version(self, device_location):
+        """
+        To be overridden by subclass
+        """
+        raise NotImplementedError
+
+    def get_circuitpython_version(self, device_location):
+        """
+        Returns the version number of CircuitPython running on the board connected
+        via ``device_url``, along with the board ID.
+
+        :param str device_location: http based device URL or local file path.
+        :return: A tuple with the version string for CircuitPython and the board ID string.
+        """
+        return self._get_circuitpython_version(device_location)
```

Comment:
Why have this wrapper instead of just overriding the public version?

Reply:
no good reason I think, just a leftover quirk from the old version of the code where they had different names originally. When I combined them to the same name it didn't occur to me to cut out the middle layer entirely.